pkg/flow/internal/controller/component_node_manager.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/grafana/river/ast"
+)
+
+// ComponentNodeManager is responsible for creating new component nodes and
+// obtaining the necessary information to run them.
+type ComponentNodeManager struct {
+	globals ComponentGlobals
+	// builtinComponentReg returns information to build and run built-in components.
+	builtinComponentReg ComponentRegistry
+
+	mut sync.RWMutex
+	// customComponentReg returns information to build and run custom components.
+	customComponentReg *CustomComponentRegistry
+}
+
+// NewComponentNodeManager creates a new ComponentNodeManager without custom component registry.
+func NewComponentNodeManager(globals ComponentGlobals, componentReg ComponentRegistry) *ComponentNodeManager {
+	return &ComponentNodeManager{
+		globals:             globals,
+		builtinComponentReg: componentReg,
+	}
+}
+
+// CreateComponentNode creates a new builtin component or a new custom component.
+func (m *ComponentNodeManager) createComponentNode(componentName string, block *ast.BlockStmt) (ComponentNode, error) {
+	// firstPart may correspond either to a namespace or to a componentName for custom components
+	firstPart := strings.Split(componentName, ".")[0]
```

Comment:
The componentName argument is set by calling `block.GetBlockName()`, so it seems redundant to split it into parts again. Can this be replaced with `block.Name[0]`?